src/lib/transaction_inclusion_status/transaction_inclusion_status.ml

```diff
@@ -36,29 +36,24 @@ let get_status ~frontier_broadcast_pipe ~transaction_pool cmd =
       let in_breadcrumb breadcrumb =
         breadcrumb |> Transition_frontier.Breadcrumb.validated_transition
         |> Mina_block.Validated.valid_commands
-        |> List.exists ~f:(fun { data = cmd'; _ } ->
-               User_command.equal cmd (User_command.forget_check cmd') )
+        |> List.exists ~f:(fun { data = found; _ } ->
+               let found' = User_command.forget_check found in
+               (* We do equality check w/o proofs and hashes first to ensure
```

Comment:
This says `first`, but there is no 'later' when we check equality with proofs. It's likely that this comment can just be deleted, or we can discuss the merits of doing a check with proof / hash equalities. Either way, it doesn't seem necessary to check the hash equalities: if the contents match, then the hashes ought to as well.